pkg/provision/automount/common.go

```diff
@@ -59,8 +59,12 @@ func ProvisionAutoMountResourcesInto(podAdditions *v1alpha1.PodAdditions, api sy
 	}
 
 	for idx, initContainer := range podAdditions.InitContainers {
-		podAdditions.InitContainers[idx].VolumeMounts = append(initContainer.VolumeMounts, resources.VolumeMounts...)
-		podAdditions.InitContainers[idx].EnvFrom = append(initContainer.EnvFrom, resources.EnvFromSource...)
+		// Don't add automount resources if persistent home init container exists, because
+		// automount resources can conflict with the init container's stow command
+		if !persistentHome || initContainer.Name != constants.HomeInitComponentName {
```

Comment:
The way this is written is fine, but it might read a bit easier if we used `continue`  to skip the homeInitComponent. Something like:

```go
for idx, initContainer := range podAdditions.InitContainers {

	// Don't add automount resources if persistent home init container exists, because
	// automount resources can conflict with the init container's stow command
	if persistentHome && initContainer.Name == constants.HomeInitComponentName {
		continue
	}

	podAdditions.InitContainers[idx].VolumeMounts = append(initContainer.VolumeMounts, resources.VolumeMounts...)
	podAdditions.InitContainers[idx].EnvFrom = append(initContainer.EnvFrom, resources.EnvFromSource...)
}
```

This is just a suggestion, however, not a necessary change.